src/main/java/Duke.java

```diff
@@ -1,10 +1,232 @@
+
+    public Duke() {
+        this.tasks = readTasksFromHardDrive();
+        this.numOfTasks = this.tasks.size();
+    }
+
+    private String taskTense() {
+        return numOfTasks == 1 ? " task" : " tasks";
+    }
+
+    private void greet() {
+        System.out.println(line);
+        System.out.println("Hi there! I'm Duke\n" +
+                "     What's up?");
+        System.out.println(line);
+    }
+
+    private void echo(String input) throws DukeException {
+        throw new DukeException("What's " + input + " ??" + "\nPlease say something I can understand!");
+    }
+
+    private void listTasks() {
+        if (numOfTasks == 0) {
+            System.out.println("Nothing to do right now...");
+        } else {
+            System.out.println("Tasks: ");
+            System.out.println(convertTasksToListString());
+        }
+        System.out.println("You have " + numOfTasks + taskTense() + "!");
+    }
+
+    private void changeTaskStatus(int index, boolean bool) throws DukeException {
+        try {
+            if (bool) {
+                tasks.get(index).markAsDone();
+                System.out.println("Good job! I've marked this task as done:");
+                System.out.println(tasks.get(index));
+            } else {
+                tasks.get(index).markAsUndone();
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println(tasks.get(index));
+            }
+            saveToHardDisk();
+        } catch (Exception e) {
+            throw new DukeException("No such task!");
+        }
+    }
+
+    private void addTask(String input) throws DukeException {
+        String description = input.substring(input.indexOf(" ") + 1);
+        if (input.startsWith("todo")) {
+            String[] command = input.split("todo", 2);
+            if (command.length < 2 || command[1].trim().length() == 0) {
+                throw new DukeException("Hold up! Description cannot be empty!");
+            }
+            ToDo newTodo = new ToDo(description);
+            tasks.add(numOfTasks, newTodo);
+            numOfTasks++;
+        } else if (input.startsWith("deadline")) {
+            String[] command = input.split("deadline", 2);
+            String[] date = command[1].split("/by", 2);
+            if (command[1].trim().length() == 0 || date[0].trim().length() == 0) {
+                throw new DukeException("Hold up! Description cannot be empty!");
+            }
+            if (date.length < 2 || date[1].trim().length() == 0) {
+                throw new DukeException("Wait! When do you want to do this by??");
+            }
+            Deadline newDeadline = new Deadline(description.substring(0, description.indexOf("/by") - 1),
+                    description.substring(description.indexOf("/") + 4));
+            tasks.add(numOfTasks, newDeadline);
+            numOfTasks++;
+        } else if (input.startsWith("event")) {
```

Comment:
This method is quite long. Perhaps you could have other helper methods to reduce the number of lines in this method? Creating `Ui` and `Parser` class abstractions might help!